src/main/java/com/api/dto/Engines.java

```diff
@@ -4,10 +4,12 @@
 import lombok.AllArgsConstructor;
 import lombok.Getter;
 import lombok.NoArgsConstructor;
+import lombok.Setter;
 
 @AllArgsConstructor
 @NoArgsConstructor
 @Getter
+@Setter
```

Comment:
why do you need the setters?

Reply:
when is called endpoint to get entities from the database, JPA is converted to DTO through the mapper without @Setter mapper cannot set values to DTO from JPA